src/kakarot/kakarot.cairo

```diff
@@ -273,10 +273,10 @@ func write_account_bytecode{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, rang
 // @param evm_address The evm address of the account.
 @external
 func upgrade_account{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
-    evm_address: felt
+    evm_address: felt, new_class_hash: felt
```

Comment:
updated to take the new_class_hash instead of taking it from account_class.

There is a semantics problem after this PR, because:
- uninitialized_account becomes the default `account_proxy`
- account_contract_class is actually just the `account_class`

when setting `set_account_contract_class` we're setting the `account_class` that should be use to execute instructions passed to the account proxy

thus, it should never be updated to contain the account proxy class (which was the source of the issue in the tests before)